### ISSUE TITLE
use a template engine instead of NOT INVENTED HERE

### ISSUE THREAD

**Issue by jappeace** [jappeace/offertex#14]
This is especially important for the table generation part, which is now done within the main code base but with large latex generation parts (rather than pure python).

# Posibilities:

## Jinjia
http://jinja.pocoo.org/ looked nice.

However, this would require a major rethink of the current engine probably. We first need to study carefully how jinja works and see if this is possible.

## Komascript
Alternatively we could use komascript, but I found it really hard to understand.

**Comment by jappeace** [comment]
After revisiting a komascript I decided that using that would be terrible. Its extremely hard to read.

Therefore we chose to use Jinjia instead. Development will be done on the template engine branch.
Several steps have to be undertaken:

Firstly we'll translate the existing template into a valid jinjia template. Note that this will be just a syntactical translation, rendering of the tables etc should be done in another issue (because that will be more involved).
With the jinjia template we'll try and give a list of the vars such as described [here](https://stackoverflow.com/questions/8260490/how-to-get-list-of-all-variables-in-jinja-2-templates).
And then we make it work.

Once this is finished we can merge back into the master branch and see if the clients work with the relatively shallow jinjia implementation (maybe it will be more difficult to use jinjia on windows).
